pkg/agent/agent.go

```diff
@@ -845,6 +848,15 @@ func (i *Initializer) allocateGatewayAddresses(localSubnets []*net.IPNet, gatewa
 	if err := util.ConfigureLinkAddresses(i.nodeConfig.GatewayConfig.LinkIndex, gwIPs); err != nil {
 		return err
 	}
+	// Periodically check whether IP configuration of the gateway is correct.
+	// Terminated when stopCh is closed.
+	if !i.networkConfig.TrafficEncapMode.IsNetworkPolicyOnly() {
+		go wait.Until(func() {
+			if err := util.ConfigureLinkAddresses(i.nodeConfig.GatewayConfig.LinkIndex, gwIPs); err != nil {
+				klog.Errorf("Failed to check IP configuration of the gateway: %v", err)
+			}
+		}, 60*time.Second, i.stopCh)
```

Comment:
It's a bit weird to start that long-lived goroutine in the agent initializer. @tnqn do you think there is an elegant way to have that code be in the route client?

Reply:
The code was put in route client but was written with some code redundant with agent initializer, so I suggested to move it to where it's first called as it requires less code and context to start the goroutine, e.g. calcuating the gwIPs, checking the traffic mode. Also, route client doesn't do the initialization so it may be a little surprising if it does the periodical sync, and the IP configuration would be managed two modules.

Maybe we could think the initializer in this way: it initializes the goroutine that syncs the gateway IP? BTW, we already start a long-lived goroutine in it to replay flows upon reconnection.

Reply:
sounds good, we can keep it this way